hs_core/models.py

```diff
@@ -961,6 +961,36 @@ def __unicode__(self):
         return self.value
 
 
+class FundingAgency(AbstractMetaDataElement):
+    term = 'FundingAgency'
+    agency_name = models.TextField(null=False)
+    award_title = models.TextField(null=True, blank=True)
+    award_number = models.TextField(null=True, blank=True)
+    agency_url = models.URLField(null=True, blank=True)
+
+    class Meta:
+        unique_together = ("content_type", "object_id", "award_number", "agency_name")
```

Comment:
Not all awards have numbers. You could have two awards from the same agency if they have different titles or if they have different award numbers.